ambry-api/src/main/java/com.github.ambry/rest/RestUtils.java

```diff
@@ -562,6 +562,17 @@ public static String getOperationOrBlobIdFromUri(RestRequest restRequest, RestUt
       endIndex = endIndex - subResource.name().length() - 1;
     }
     String operationOrBlobId = path.substring(startIndex, endIndex);
+    // remove signed path if present and if it's not a signed id
```

Comment:
This part is a little complicated, or maybe there is a better implementation. The code was written based following considerations:
1. Current URIs (i.e. get blob/ get BlobInfo/ signed id) should still work correctly without any errors;
2. When `Signed Path` is embedded in URI, this method can still extract BlobId correctly;
3. The logic in this method is, after removing query string/prefix/subResource, if there are more than 2 segments ( or 3 if leading slash exists), the first non-empty segment is signed path. 
4. This is a little controversial because it requires `getOperation` like GET_PEERS etc only occupies one segment. The URI like `/peers/random/extra` is not allowed any more, otherwise, the first segment will be considered signed path and be removed. 
5. For 4, I currently don't have a better way to address this. Let me know if you have good approach.